python/benchmark/applications/scalability_tests/test.py

```diff
@@ -0,0 +1,135 @@
+import importlib
+import os
+from benchmark.common.base_test_suite import BaseTestSuite, TestCase, register
+from benchmark.common.base_test_group import BaseTestGroup
+from benchmark.applications.scalability_tests.scalable_base_test_group import ScalableTestGroup
+from benchmark.common.simulation import Simulation
+from benchmark.instrumented.instrumented_frame import InstrumentedFrame
+from benchmark.instrumented.instrumented_connectors import \
+    InstrumentedSpacetimeConnection, \
+    InstrumentedObjectlessSpacetimeConnection, \
+    InstrumentedMySqlConnection
+from spacetime.client.declarations import Producer, GetterSetter, Tracker,\
+    Deleter, Setter, Getter
+
+MAX_NUM_SCALING_CLIENTS = 10000
+
+
+class ScalableTestSuite(BaseTestSuite):
+    def make_scalable_test_group(self, testcase):
+        filename = "Server_{0}".format(testcase.test_name)
+        server_instrument_filename = os.path.join(self.foldername, filename)
+        groupname = testcase.test_name
+        print("TEST: 2", self.server)
+        scalable_test_group = ScalableTestGroup(
+            groupname, self.logger, self.server, self.foldername,
+            server_instrument_filename)
+        app_id_m = "Master_{0}".format(groupname)
+        app_id_c = "Client_{0}".format(groupname)
+
+        connector_m = self.connector(app_id_m, **self.kwargs)
+        connector_c = self.connector(app_id_c, **self.kwargs)
+
+        frame_m = InstrumentedFrame(connector_m, time_step=self.args.timestep)
+        frame_c = InstrumentedFrame(connector_c, time_step=self.args.timestep)
+
+        combination_m = [
+            Producer(*testcase.test_module.BM_PRODUCER),
+            Getter(*testcase.test_module.BM_GETTER),
+            Setter(*testcase.test_module.BM_SETTER),
+            GetterSetter(*testcase.test_module.BM_GETTERSETTER),
+            Tracker(*testcase.test_module.BM_TRACKER),
+            Deleter(*testcase.test_module.BM_DELETER)]
+        combination_c = [
+            Producer(*testcase.test_module.BT_PRODUCER),
+            Getter(*testcase.test_module.BT_GETTER),
+            Setter(*testcase.test_module.BT_SETTER),
+            GetterSetter(*testcase.test_module.BT_GETTERSETTER),
+            Tracker(*testcase.test_module.BT_TRACKER),
+            Deleter(*testcase.test_module.BT_DELETER)]
+
+        foldername_m = os.path.join(
+            self.foldername, "Producer." + testcase.test_name)
+        foldername_c = os.path.join(
+            self.foldername, "Consumer." + testcase.test_name)
+
+        master = Simulation(
+            frame_m, testcase.instances, testcase.steps,
+            testcase.test_module.initialize, testcase.test_module.update,
+            combination_m, foldername_m, app_id_m)
+        client_gen = (
+            Simulation(
```

Comment:
Create a new function that creates a new `frame_c`, `connector_c`, `combination_c` and a new `Simulation` with that and return it to this generator. (Otherwise each sim is using the same frame object. That's a recipe for disaster.